tracing-subscriber/src/fmt/fmt_layer.rs

```diff
@@ -0,0 +1,457 @@
+use crate::{
+    field::RecordFields,
+    fmt::{format, FormatEvent, FormatFields, MakeWriter},
+    layer::{Context, Layer},
+    registry::{LookupSpan, SpanRef},
+};
+use std::{any::TypeId, cell::RefCell, fmt, io, marker::PhantomData};
+use tracing_core::{
+    span::{Attributes, Id, Record},
+    Event, Subscriber,
+};
+
+/// A [`Layer`] that logs formatted representations of `tracing` events.
+///
+/// [`Layer`]: ../trait.Layer.html
+#[derive(Debug)]
+pub struct FmtLayer<
+    S,
+    N = format::DefaultFields,
+    E = format::Format<format::Full>,
+    W = fn() -> io::Stdout,
+> {
+    make_writer: W,
+    fmt_fields: N,
+    fmt_event: E,
+    _inner: PhantomData<S>,
+}
+
+/// A builder for [`FmtLayer`](struct.FmtLayer.html) that logs formatted representations of `tracing`
+/// events.
+#[derive(Debug)]
+pub struct Builder<
+    S,
+    N = format::DefaultFields,
+    E = format::Format<format::Full>,
+    W = fn() -> io::Stdout,
+> {
+    fmt_fields: N,
+    fmt_event: E,
+    make_writer: W,
+    _inner: PhantomData<S>,
+}
+
+impl<S> FmtLayer<S> {
+    /// Returns a new [`Builder`](struct.Builder.html) for configuring a `FmtLayer`.
+    pub fn builder() -> Builder<S> {
+        Builder::default()
+    }
+}
+
+// This, like the MakeWriter block, needs to be a seperate impl block because we're
+// overriding the `E` type parameter with `E2`.
```

Comment:
Yeah, that shouldn't be true. It just means you can't use `-> Self`, but that's fine.

Reply:
@davidbarsky That's correct, the separate impl blocks are necessary because they place different bounds on the type parameters, not because of changing a type param in a return type. I'm fine with leaving all the impl blocks as-is in this PR (although I would prefer the functions with the same bounds be grouped together) but I really think we should remove the comment, since it's factually incorrect.